sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
-    complex = Predicate('complex')
-    composite = Predicate('composite')
-    even = Predicate('even')
-    extended_real = Predicate('extended_real')
-    hermitian = Predicate('hermitian')
-    imaginary = Predicate('imaginary')
-    infinitesimal = Predicate('infinitesimal')
-    infinite = Predicate('infinite')
-    integer = Predicate('integer')
-    irrational = Predicate('irrational')
-    rational = Predicate('rational')
-    algebraic = Predicate('algebraic')
-    transcendental = Predicate('transcendental')
-    negative = Predicate('negative')
-    nonzero = Predicate('nonzero')
-    positive = Predicate('positive')
-    prime = Predicate('prime')
-    real = Predicate('real')
-    odd = Predicate('odd')
-    is_true = Predicate('is_true')
-    nonpositive = Predicate('nonpositive')
-    nonnegative = Predicate('nonnegative')
-    zero = Predicate('zero')
-
-    symmetric = Predicate('symmetric')
-    invertible = Predicate('invertible')
-    singular = Predicate('singular')
-    orthogonal = Predicate('orthogonal')
-    unitary = Predicate('unitary')
-    normal = Predicate('normal')
-    positive_definite = Predicate('positive_definite')
-    upper_triangular = Predicate('upper_triangular')
-    lower_triangular = Predicate('lower_triangular')
-    diagonal = Predicate('diagonal')
-    triangular = Predicate('triangular')
-    unit_triangular = Predicate('unit_triangular')
-    fullrank = Predicate('fullrank')
-    square = Predicate('square')
-    real_elements = Predicate('real_elements')
-    complex_elements = Predicate('complex_elements')
-    integer_elements = Predicate('integer_elements')
-
-    @classproperty
+from sympy.utilities.decorator import memoize_property
+
+
+# Deprecated predicates should be added to this list
+deprecated_predicates = [
+    'bounded',
+    'infinity'
+]
+
+# Memoization storage for predicates
+predicate_storage = {}
+
+
+class QClass(object):
+    """
+    This class contains all the supported keys by ``ask``.
+    """
+
+    @memoize_property(predicate_storage)
+    def hermitian(self):
+        """
+        Hermitian predicate.
+
+        ``ask(Q.hermitian(x))`` is true iff ``x`` belongs to the field of
+        hermitian operators.
+
+        TODO: Add examples
+
+        """
+        return Predicate('hermitian')
+
+    @memoize_property(predicate_storage)
+    def antihermitian(self):
+        """
+        Antihermitian predicate.
+
+        ``Q.antihermitian(x)`` is true iff ``x`` belongs to the field of
+        antihermitian operators, i.e., operators in the form ``x*I``, where
+        ``x`` is Hermitian.
+
+        TODO: Add examples
+
+        """
+        return Predicate('antihermitian')
+
+    @memoize_property(predicate_storage)
+    def real(self):
+        r"""
+        Real number predicate.
+
+        ``Q.real(x)`` is true iff ``x`` is a real number, i.e., it is in the
+        interval :math:`(-\infty, \infty)`.  Note that, in particular the infinities
```

Comment:
I don't think the `:math:` is necessary for inline math, and it makes it a little easier to read in plain text without it.